### PR TITLE
[FIX] Prevent initialisation of the graphical GUI on WSL

### LINKED ISSUE #71: Change input.py to run under Linux WSL
## Expected Behavior
The pass (see further down) should be replaced by a check to see if command line parameters were supplied else it errors out later on (wsl does not have a graphical interface (yet))

## Current Behavior
running "python3 wahoo_map_creator.py" using Linux WSL under Windows
grafik

## Steps to Reproduce the Issue

1. run "python3 wahoo_map_creator.py" using Linux WSL under Windows

## Context
 * OS/Platform: `Linux WSL under Windows`
 * Other information regarding your environment *(optional)*: `nothing`

## Log Output / Stack Trace

get it to run in Windows using WSL:

- Needed to do this sudo apt-get install zip unzip
- and sudo apt-get install python3.8-tk (python version matters)
- Install osmium (and pyosmium) with pip install
- Changed input.py with

```
         try:
            tk.Tk.__init__(self, *args, **kwargs)

            # self.geometry("420x360")
            self.title("Wahoo map creator")
            self.configure(bg="white")
            self.option_add("*Font", "Calibri 16")

            container = tk.Frame(self)
            container.pack(side = tk.TOP, fill = tk.BOTH, expand = True)

            self.first = ComboboxesEntryField(container, self.o_input_data.max_days_old)
            self.first.pack(side = tk.TOP, fill = tk.X)

            self.third = Checkbuttons(container, self.o_input_data, controller = self)
            self.third.pack(side = tk.TOP, fill = tk.X)

            self.four = Buttons(container, controller = self)
            self.four.pack(side = tk.TOP, fill = tk.X)
        except:
            pass
```

### PR DESCRIPTION
**Pull request by Ebe66**
## This PR…

Prevents the initialisation of the graphical GUI on  WSL.

closes #71

## Considerations and implementations

If running the programm via "python3 wahoo_map_creator.py" but not having a GUI available, tk stuff can not be processed.

## Pull Request Checklist
- [x] Reviewed the [Contributing Guidelines](https://github.com/treee111/wahooMapsCreator/blob/develop/.github/CONTRIBUTING.md)
    + Especially the [Pull-Requests](https://github.com/treee111/wahooMapsCreator/blob/develop/.github/CONTRIBUTING.md#Pull-Requests) section
- [x] commits (and commit-messages) are understandable
- [x] {...}

### CODE REVIEW

File: common_python/input.py [hunk 1]
```diff
@@ -56,6 +57,10 @@ def __init__(self, *args, **kwargs):
 
         self.o_input_data = InputData()
 
+        if 'microsoft' in uname().release:
+            self.gui_mode = False
+            return
+
         tk.Tk.__init__(self, *args, **kwargs)
```
Comment (treee111) [inline common_python/input.py:64]: would it help to move lines 64 until 83 into the if-statement after line 54?
line 58 needs then to be the first line of this init-function

Or even move lines 66 until 83 into the start_gui() function and leave 64 in the if-statement after line 54?

for macOS that seams to work. I wonder if it also works on WSL?

**Review by treee111 — CHANGES_REQUESTED** [review]
thanks for this PR! 👍 
you nicely spotted a wrong thing here in my opinion!
The inititalization of the GUI should not happen always. That should happen only if in GUI mode (and then, a GUI and tk is available)

I added a inline-comment to a possible other solution to takkle that.

File: common_python/input.py (hunk 1 above)
Comment (treee111) [reply common_python/input.py:64]: as discussed, this is an other topic which is tackled in #73. this here is good to go! 💯

**Review by treee111 — APPROVED** [review]

**Comment by mbrieske** [comment]
May I ask what bug this PR actually fixes? I work mainly on Ubuntu inside WSL for most of my programming stuff, and after installing wahoomc when I first wanted to start the Gui the error "no display found" drove me crazy. That is, until I found out about this "bugfix"... removed it, and now everything is working fine.

**Comment by Ebe66** [comment]
Hope Treee111 does not mind me trying to answer this.

At the time of this PR wsl did not, to our knowledge, have support for graphical interfaces. So a check was added to see if we where running on wsl and if so abort the gui.
It appears the graphical interface was added in the mean time. I'm watching a youtube on it now.    
@treee111 maybe we can do some testing and (re)add it. Maybe it is possible to check the wsl version used?

**Pull request merged**